### PR TITLE
inbox and deleted message pages uplift

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [x] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [x] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [x] Code does not use .html_safe
- [x] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187142002
Ticket: https://www.pivotaltracker.com/story/show/187142029

# A brief description of the changes

Current behavior: Messages and Deleted Messages pages needed uplifts to bootstrap format, message viewing outdated

New behavior: All views improved and more closely resembling figma designs

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

Screenshot 2024-07-16 at 6 22 41 PM
Screenshot 2024-07-16 at 6 22 45 PM
Screenshot 2024-07-16 at 6 22 52 PM
Screenshot 2024-07-16 at 6 22 58 PM
Screenshot 2024-07-16 at 6 23 04 PM

### CODE REVIEW

**Comment by ATBull81** [comment]
Good to review, just have one little snag to still fix with getting nav tabs to line up properly (see zoomed in photo below)

Screenshot 2024-07-16 at 6 22 13 PM

File: app/controllers/inboxes_controller.rb
Comment (charlienparker) [inline app/controllers/inboxes_controller.rb:1]: is the progress layout unneeded for these actions?

File: app/javascript/css/main.scss [hunk 1]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+.delete-icon {
+  color: var(--status-red);
+}
+
+.inbox-sender {
+  font-size: 16px;
+  font-weight: 600;
+}
+
+.tabs {
```
Comment (charlienparker) [inline app/javascript/css/main.scss:528]: can we use bootstrap for these margins?

File: app/javascript/css/main.scss [hunk 2]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+  margin-top: 3ch;
+}
+
+.tab {
```
Comment (charlienparker) [inline app/javascript/css/main.scss:532]: can we use bootstrap for the border rounding, margins, padding, and border?

File: app/javascript/css/main.scss [hunk 3]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+.tab.active {
+  background: #ffffff;;
+  color: var(--primary-color);
+  border-top: 1px solid #dee2e6;
+  border-left: 1px solid #dee2e6;
+  border-right: 1px solid #dee2e6;
+  border-bottom: none;
+}
+
+.no-underline {
```
Comment (charlienparker) [inline app/javascript/css/main.scss:553]: can we use bootstrap for the text decoration?

File: app/javascript/css/main.scss [hunk 4]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+  text-decoration: none;
+}
+
+.no-underline:hover {
+  text-decoration: none;
```
Comment (charlienparker) [inline app/javascript/css/main.scss:558]: can we use bootstrap for the text decoration?

File: app/javascript/css/main.scss [hunk 5]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+}
```
Comment (charlienparker) [inline app/javascript/css/main.scss:559]: Leverage scss nesting:

```
.no-underline {
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}

```

Although I'm not even sure we need a hover pseudoclass?

File: app/javascript/css/main.scss [hunk 6]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+
+.tab.active {
+  background: #ffffff;;
+  color: var(--primary-color);
+  border-top: 1px solid #dee2e6;
+  border-left: 1px solid #dee2e6;
+  border-right: 1px solid #dee2e6;
+  border-bottom: none;
+}
+
```
Comment (kristinmerbach) [inline app/javascript/css/main.scss:552]: tab borders don't quite line up. will have updated css for you

File: app/views/insured/families/_inbox.html.erb [hunk 7]
```diff
@@ -1,64 +1,128 @@
...
+    <a class="no-underline tabs tab tab-content <%= 'active' if @folder.try(:capitalize) == 'Deleted' %>" href="<%= inbox_insured_families_path(@provider, folder: 'Deleted') %>">
+      <% if @folder == 'Deleted' %>
+        <strong><%= l10n("deleted") %></strong>
+      <% else %>
+        <%= l10n("deleted") %>
+      <% end %>
+    </a>
+  </div>
+  <% if @folder == 'Deleted' && sorted_inbox_messages.size == 0 %>
+    <div class="rounded border bg-white mb-4 mt-1">
```
Comment (kristinmerbach) [inline app/views/insured/families/_inbox.html.erb:27]: <div class="rounded-end bg-white mb-4 tab-container">

File: app/views/insured/families/_inbox.html.erb [hunk 8]
```diff
@@ -1,64 +1,128 @@
...
+      <section>
+        <div class="pl-2 pr-2 pt-3">
+          <p><%= l10n("zero_deleted_messages") %></p>
+        </div>
+      </section>
+    </div>
+  <% else %>
+    <div id="message_list_form" class="rounded border bg-white mb-4 mt-1">
```
Comment (kristinmerbach) [inline app/views/insured/families/_inbox.html.erb:35]: class="rounded-end bg-white mb-4 tab-container"

File: app/views/insured/families/_inbox.html.erb [hunk 9]
```diff
@@ -1,64 +1,128 @@
...
-    <div class="inboxes" id="inbox_form">
-      <div class="pull-right">
-        <a href="<%= inbox_insured_families_path(@provider) %>">
-          <% if @folder == 'Inbox' %>
-            <strong><%= l10n("inbox") %></strong>
-          <% else %>
-            <%= l10n("inbox") %>
+<% if @bs4 %>
+  <% sorted_inbox_messages = @provider.inbox.messages.select{|m| @folder == (m.folder.try(:capitalize) || 'Inbox') }.sort_by(&:created_at).reverse %>
+  <div id="inbox-headers">
```
Comment (kristinmerbach) [inline app/views/insured/families/_inbox.html.erb:3]: add class of 'tab-holder'

File: app/javascript/css/main.scss [hunk 10]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+.no-underline {
+  text-decoration: none;
+}
+
+.no-underline:hover {
+  text-decoration: none;
+}
+
+.msg-inbox:hover {
```
Comment (kristinmerbach) [inline app/javascript/css/main.scss:561]: updated tab css is below, you'll also have to add a new color of the teritary-tint
`.tab-holder {
  margin-bottom: 6px;
}

.tab {
  border: none;
  margin-right: 0.25ch;
  padding: 1.25ch 1ch 1ch;
  font-size: 1rem;
  font-weight: 600;
  color: var(--default-font-color);
  background: var(--tertiary-tint);
  border-radius: 5px 5px 0 0;
  border-bottom: 1px solid var(--grey-060);
}

.tab.active {
  background: #ffffff;;
  color: var(--primary-color);
  border-top: 1px solid var(--grey-060);
  border-left: 1px solid var(--grey-060);
  border-right: 1px solid var(--grey-060);
  border-bottom: 1px solid var(--grey-000);
}

.tab-container {
  margin-top: 5px;
  border: 1px solid var(--grey-060);
}`

**Review by kristinmerbach — CHANGES_REQUESTED** [review]
A few CSS issues, but good job!

File: app/controllers/insured/families_controller.rb [hunk 11]
```diff
@@ -696,6 +696,8 @@ def resolve_layout
     case action_name
     when "find_sep"
       EnrollRegistry.feature_enabled?(:bs4_consumer_flow) ? "progress" : "application"
+    when "inbox"
+      EnrollRegistry.feature_enabled?(:bs4_consumer_flow) ? "progress" : nil
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply app/controllers/insured/families_controller.rb:700]: Yep, getting a missing template if "" is used, getting no side nav if `application` is used. `nil` seems to be the working option, could change it to a simple ternary if preferred.
Screenshot 2024-07-17 at 11 32 34 PM

File: app/javascript/css/main.scss (hunk 1 above)
Comment (ATBull81) [reply app/javascript/css/main.scss:528]: so these were pulled from dbh repo with a little modification (https://github.com/ideacrew/enroll/pull/4031#discussion_r1682000321). On top of this, the mb-# options don't quite get to the specific measurements needed (mb-2 is 8px, which is the closest option). If there are other bootstrap classes that can get there I'd be open to it

File: app/views/insured/families/_inbox.html.erb (hunk 9 above)
Comment (ATBull81) [reply app/views/insured/families/_inbox.html.erb]: https://github.com/ideacrew/enroll/pull/4031/commits/2acbaa0d22ab53f4730d06651ec91c91b1a69c73#diff-02e0e47ade8e44ae0909e3b718d6a44798b9ba6e661c34186c82b4997b36705cR3

File: app/views/insured/families/_inbox.html.erb (hunk 8 above)
Comment (ATBull81) [reply app/views/insured/families/_inbox.html.erb]: https://github.com/ideacrew/enroll/pull/4031/commits/2acbaa0d22ab53f4730d06651ec91c91b1a69c73#diff-02e0e47ade8e44ae0909e3b718d6a44798b9ba6e661c34186c82b4997b36705cR27-R35

File: app/javascript/css/main.scss [hunk 12]
```diff
@@ -515,3 +515,54 @@ a.disabled {
...
+}
+
+.tab {
+  border: none;
+  margin-right: 0.25ch;
+  padding: 1.25ch 1ch 1ch;
+  font-size: 1rem;
+  font-weight: 600;
+  color: var(--grey-140);
+  background: var(--grey-020);
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply app/javascript/css/main.scss]: color added (but then renamed) here: https://github.com/ideacrew/enroll/pull/4031/commits/9e72ffcb7b0c91d3aec7dfea07fbdc6339a29fec#diff-5262cdd01f4641dc5ed634a7a071f3f7925aee067896706e2d9ed93c479f58d4R61

File: app/javascript/css/main.scss (hunk 2 above)
Comment (ATBull81) [reply app/javascript/css/main.scss:532]: I think I lean leaving them all in the dbh css classes so it's clearly laid out how to get the tab look, but if it's strongly preferred I can change

File: app/controllers/insured/families_controller.rb (hunk 11 above)
[the remark this answers is not in the document]
Comment (ATBull81) [reply app/controllers/insured/families_controller.rb]: https://github.com/ideacrew/enroll/pull/4031/commits/9b23f39fd06f275d8c80fd3eb843874d94dcf690 pushed up. two_column change, works as intended for dc view

File: app/views/shared/inboxes/_individual_message.html.erb [hunk 13]
```diff
@@ -1,23 +1,56 @@
-<tr tabindex="0" onkeydown="handleButtonKeyDown(event, 'message_inbox')", id="message_inbox" class='<%=message.message_read==true ? "msg-inbox-read" : "msg-inbox-unread"%>' onclick="showMessage('<%= retrieve_show_path(provider, message) %>');">
-  <td style="font-weight:bold">
-    <%= message.try(:from) if @folder == 'Inbox' %>
-    <%= message.try(:to)   if @folder == 'Sent'%>
-  </td>
-  <td>
-    <%=message.created_at.in_time_zone("Eastern Time (US & Canada)").strftime("%m/%d/%Y") %>
+<% if @bs4 %>
+  <tr tabindex="0" onkeydown="handleButtonKeyDown(event, 'message_inbox')", id="message_inbox" class='msg-inbox <%=message.message_read==true ? "msg-inbox-read" : "msg-inbox-unread"%>' onclick="showMessage('<%= retrieve_show_path(provider, message) %>');">
```
Comment (charlienparker) [inline app/views/shared/inboxes/_individual_message.html.erb:2]: I think we should move the onclick into a script

Comment (ATBull81) [reply app/views/shared/inboxes/_individual_message.html.erb]: https://github.com/ideacrew/enroll/pull/4031/commits/965ebe23df3e72e386a566f00cba30c3344630f7#diff-85d9f92e5d3bed01e04096dd5fdfd404a78d94aea9a28076f504c8bd119230e0R2-R42

**Review by kristinmerbach — APPROVED** [review]
LGTM!

**Pull request merged**